docs/development/module/fees.md

```diff
@@ -1,50 +1,126 @@
 ---
-title: Transaction Fees
+title: Transaction Weight and Fees
 ---
 
-When transactions are submitted to a blockchain, they are executed by the nodes in the network. To be economically sustainable, the cost nodes incur to execute a transaction must be covered by the submitter of the transaction. The cost to execute transactions varies over orders of magnitude, and thus Substrate provides a flexible mechanism for characterizing the total cost of a transaction.
+When transactions are submitted to a blockchain, they are executed by the nodes in the network. To
+be economically sustainable, the cost nodes incur to execute a transaction must be covered by the
+submitter of the transaction. The cost to execute transactions could vary over orders of magnitude,
+and thus Substrate provides a flexible mechanism for characterizing the total, presumably minimum,
+cost of a transaction in order to be included into a block.
+
+Furthermore, a number of resources in any chain can be limited by nature such as time, memory and
+computation. A mechanism should exist to prevent individual resource-consuming components of the
+chain, including dispatchables, to consume too much of any of the mentioned resources. This concept
+is represented in substrate by weights.
+
+## Weights Basics
+
+The fee system is dependent on the weight system. Hence, we will first briefly describe the weights.
+
+As mentioned, weights can technically refer to, or represent, numerous _limited_ resources. However,
+at the time of this writing, substrate weights are simply a numeric value (todo: link). Each
+dispatchable function is given a weight using the `#[weight = $x]` annotation, where `$x` is some
+function capable of determining the weight of the dispatch. `$x` can access and examine the
+arguments of the dispatch. Nonetheless, a weight calculation should always:
+- Be computable __ahead of dispatch__. A block producer, or a validator, should be able to examine
```

Comment:
So this is why it doesn't make sense to use storage values in the calculation.

Reply:
Yes I have not made that more clear.